srtcore/window.cpp

```diff
@@ -158,99 +158,44 @@ void srt::CPktTimeWindowTools::initializeWindowArrays(int* r_pktWindow, int* r_p
 }
 
 
-int srt::CPktTimeWindowTools::getPktRcvSpeed_in(const int* window, int* replica, const int* abytes, size_t asize, size_t hdr_size, int& bytesps)
+int srt::CPktTimeWindowTools::getPktRcvSpeed_in(const int* window, int* replica, const int* abytes, size_t asize, size_t hdr_size, int& w_bytesps)
 {
-   // get median value, but cannot change the original value order in the window
-   std::copy(window, window + asize, replica);
-   std::nth_element(replica, replica + (asize / 2), replica + asize);
-   //std::sort(replica, replica + asize);
-   int median = replica[asize / 2];
-
-   unsigned count = 0;
-   int sum = 0;
-   int upper = median << 3;
-   int lower = median >> 3;
-
-   bytesps = 0;
-   unsigned long bytes = 0;
-   const int* bp = abytes;
-   // median filtering
-   const int* p = window;
-   for (int i = 0, n = (int)asize; i < n; ++ i)
-   {
-      if ((*p < upper) && (*p > lower))
-      {
-         ++ count;  //packet counter
-         sum += *p; //usec counter
-         bytes += (unsigned long)*bp;   //byte counter
-      }
-      ++ p;     //advance packet pointer
-      ++ bp;    //advance bytes pointer
-   }
-
-   // claculate speed, or return 0 if not enough valid value
-   if (count > (asize >> 1))
-   {
-      bytes += (hdr_size * count); //Add protocol headers to bytes received
-      bytesps = (int)ceil(1000000.0 / (double(sum) / double(bytes)));
-      return (int)ceil(1000000.0 / (sum / count));
-   }
-   else
-   {
-      bytesps = 0;
-      return 0;
-   }
+    PassFilter<int> filter = GetPeakRange(window, replica, asize);
+
+    unsigned count = 0;
+    int sum = 0;
+
+    w_bytesps = 0;
+    unsigned long bytes = 0;
+    // // (explicit specialization due to problems on MSVC 2013 and 2015)
+    AccumulatePassFilterParallel<unsigned, unsigned long>(window, window + asize, filter, abytes,
```

Comment:
`window + asize` as an end pointer might trigger warnings on some static analyzers. Using just length might be better.
Just a thought.

Reply:
Well, provided that this is taken as an array anyway (not as a generic container), it should work.